### PR TITLE
[story] QuotaUI Fix validation Issue 1675

### PR DESCRIPTION
**Pull request by mvdr**
Pull request for [PBI 59622](https://niposoftware.visualstudio.com/DefaultCollection/Nfield/_workitems/edit/59622)

New validations, to match the same restrictions that are in the GLU Client

AB#59622

### CODE REVIEW

File: Nfield.Quota/QuotaFrameValidator.cs [hunk 1]
```diff
@@ -399,13 +437,126 @@ public QuotaFrameValidator()
...
 
+        private static bool HaveNestedMinLevelsSumToLessThanMaxTargetForEachLevel(
+            QuotaFrame frame,
+            IEnumerable<QuotaFrameVariable> variables,
+            PropertyValidatorContext context)
+        {
+            bool ProcessLevel(QuotaFrameVariable variable, IEnumerable<QuotaFrameLevel> parents)
+            {
+                // sum of targets, ignoring null values
+                var sum = variable.Levels.Aggregate(0, (total, level) => total + (level.Target ?? 0));
```
Comment (WimDoze) [inline Nfield.Quota/QuotaFrameValidator.cs]: Thisis not correct for multi variables.

File: Nfield.Quota/QuotaFrameValidator.cs [hunk 2]
```diff
@@ -399,13 +437,126 @@ public QuotaFrameValidator()
...
+                        allTargetsNull = false;
+                        sum += level.MaxTarget.Value;
+                    }
+                }
+
+                var isValid = true;
+
+                if (!allTargetsNull)
+                {
+                    foreach (var parent in parents)
```
Comment (WimDoze) [inline Nfield.Quota/QuotaFrameValidator.cs]: Testing more then one parent level is wrong

Comment (joukevandermaas) [reply Nfield.Quota/QuotaFrameValidator.cs]: Why is this wrong? The issue is that you can create an incorrect frame if you leave out targets for variables that are nested but also have children.

Comment (WimDoze) [reply Nfield.Quota/QuotaFrameValidator.cs]: The wrong is on the foreach.
For more then one parent level up the target can be higher then the sum of the MaxTargets in the levels.

**Pull request merged**